cc/toolchains/llvm/cc_toolchain_config.bzl

```diff
@@ -28,6 +28,8 @@ def cc_toolchain_config(
         target_system_name,
         builtin_sysroot = None,
         is_darwin = False):
+    cross_compile = host_system_name != target_system_name
```

Comment:
This is a little dangerous.

We don't enforce any sort of convention around what `host_system_name` and `target_system_name` are.

This led to `-stdlib=libstdc++` being selected because they didn't match in the `x86_64-linux` config, even though we weren't cross compiling.

What we should probably do is define a set of constants for the clang target triplets (`x86_64-linux-unknown-linux-gnu`, `aarch64-unknown-linux-gnu`, etc..)

Then in this function we should write a quick check to make sure we're using one of those constants, otherwise it's an error.

That would make this a lot more robust.